### ISSUE TITLE
Snapshot operation stuck, delete command doesn't work

### ISSUE THREAD

**Issue by vanga** [elastic/elasticsearch#10564]
We take snpashots every hour to AWS S3 directly using cloud-aws plugin
snapshot was stuck because of a reboot on master node during the snapshot, ES version 1.4.0, ArchLinux
Cluster was later upgraded to 1.4.2 at which point we noticed that there is a snapshot stuck
while trying to create a new snapshot
`{"error":"ConcurrentSnapshotExecutionException[[my_s3_repository:2015.04.13.09.16.42] a snapshot is already running]","status":503}`
curl localhost:9200/_snapshot/my_s3_repository/_all?pretty doesn't have the stuck snapshot in the list.

Delete snapshot command never completes, it hangs forever.
rolling restart (restarting one node after the other) didn't help.
Clean up script doesn't work for me https://github.com/imotov/elasticsearch-snapshot-cleanup/issues/2

In S3, there is no metadata file for the stuck snapshot, there are snapshot-<snapshot-name> files for some of the indices

How do I atleast forget/discard old snapshots..create a new repository and start fresh. I can not stop the whole cluster as it is a production cluster.

curl localhost:9200/_snapshot/_status?pretty
```
{
  "snapshots" : [ {
    "snapshot" : "2015.03.11.06.04.39",
    "repository" : "my_s3_repository",
    "state" : "ABORTED",
    "shards_stats" : {
      "initializing" : 0,
      "started" : 0,
      "finalizing" : 0,
      "done" : 17,
      "failed" : 3,
      "total" : 20
    },
    "stats" : {
      "number_of_files" : 0,
      "processed_files" : 0,
      "total_size_in_bytes" : 0,
      "processed_size_in_bytes" : 0,
      "start_time_in_millis" : 0,
      "time_in_millis" : 0
    },
    "indices" : {
      "playground_video" : {
        "shards_stats" : {
          "initializing" : 0,
          "started" : 0,
          "finalizing" : 0,
          "done" : 5,
          "failed" : 0,
          "total" : 5
        },
        "stats" : {
          "number_of_files" : 0,
          "processed_files" : 0,
          "total_size_in_bytes" : 0,
          "processed_size_in_bytes" : 0,
          "start_time_in_millis" : 0,
          "time_in_millis" : 0
        },
        "shards" : {
          "4" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "3" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "2" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "1" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "0" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          }
        }
      },
      "playground_feed" : {
        "shards_stats" : {
          "initializing" : 0,
          "started" : 0,
          "finalizing" : 0,
          "done" : 4,
          "failed" : 1,
          "total" : 5
        },
        "stats" : {
          "number_of_files" : 0,
          "processed_files" : 0,
          "total_size_in_bytes" : 0,
          "processed_size_in_bytes" : 0,
          "start_time_in_millis" : 0,
          "time_in_millis" : 0
        },
        "shards" : {
          "4" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "2" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "3" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "0" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "1" : {
            "stage" : "FAILURE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          }
        }
      },
      "playground_profile" : {
        "shards_stats" : {
          "initializing" : 0,
          "started" : 0,
          "finalizing" : 0,
          "done" : 4,
          "failed" : 1,
          "total" : 5
        },
        "stats" : {
          "number_of_files" : 0,
          "processed_files" : 0,
          "total_size_in_bytes" : 0,
          "processed_size_in_bytes" : 0,
          "start_time_in_millis" : 0,
          "time_in_millis" : 0
        },
        "shards" : {
          "2" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "1" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "4" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "3" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "0" : {
            "stage" : "FAILURE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          }
        }
      },
      "playground_admin" : {
        "shards_stats" : {
          "initializing" : 0,
          "started" : 0,
          "finalizing" : 0,
          "done" : 4,
          "failed" : 1,
          "total" : 5
        },
        "stats" : {
          "number_of_files" : 0,
          "processed_files" : 0,
          "total_size_in_bytes" : 0,
          "processed_size_in_bytes" : 0,
          "start_time_in_millis" : 0,
          "time_in_millis" : 0
        },
        "shards" : {
          "2" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "1" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "0" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "4" : {
            "stage" : "FAILURE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          },
          "3" : {
            "stage" : "DONE",
            "stats" : {
              "number_of_files" : 0,
              "processed_files" : 0,
              "total_size_in_bytes" : 0,
              "processed_size_in_bytes" : 0,
              "start_time_in_millis" : 0,
              "time_in_millis" : 0
            }
          }
        }
      }
    }
  } ]
}
```

**Comment by clintongormley** [comment]
@imotov any ideas here?

**Comment by imotov** [comment]
@clintongormley not sure I tried to reproduce the issue with cleanup script that @vanga described in https://github.com/imotov/elasticsearch-snapshot-cleanup/issues/2 but I cannot. The issue itself seems to be a duplicate of #9924 and should be fixed in 1.4.5.

**Comment by clintongormley** [comment]
thanks @imotov 

@vanga sounds like you will need to upgrade.

**Comment by vanga** [comment]
@clintongormley .. If I update to version > 1.4.5 , this issue will go away and i can start taking snapshots?

**Comment by clintongormley** [comment]
@vanga sounds like that is the case (but you'll need to upgrade to 1.5.1, as 1.4.5 has not been released yet)

**Comment by vanga** [comment]
We have joined new nodes to the clsuter with 1.5.1 version and removed old nodes. New cluster doesn't have the problem . 
Thanks.

**Comment by allthedrones** [comment]
I'm experiencing this issue in 1.6.1 ... or at least a *very* similar issue; in my case, the snapshot state (as reported by ~/_snapshot/repo/_status) is ABORTED, but the hung shard still reports a stage of "STARTED". Curiously, the output from ~/_snapshot/repo/_all does not agree with the output from .../_status -- it shows the hung snapshot as "IN_PROGRESS". Issuing a REST call to DELETE the target snapshot hangs indefinitely -- never returns, and never times out (I also couldn't find any relevant activity in pending_tasks, hot_threads, nor the log files).

For clarity, the master node did not restart, suffer a network partition, etc ... there was no interruption of any kind during the time the snapshot was running and nothing in the logs on the master node or the node with the 'stuck' shard. The repo *is* using the Azure plugin. The indices (1 per snapshot) being snapshot'd are large so the operation is long running ... typically between 30 and 90 minutes. This cluster has never run any version other than 1.6.1, and there are no other clusters pointing to this Azure container, etc.

Nothing but a full restart seems to fix this problem. I expected, based on https://github.com/elastic/elasticsearch/pull/11450, that I would be able to clean this up by simply restarting the current master and allowing one of the other two [dedicated] master nodes to be elected, but that wasn't enough to fix the problem (I tried twice for good measure). I wasn't able to get @imotov 's snapshot cleanup tool to build against 1.6.1 (maybe due to #15427?). As long as the snapshot is stuck in this state, I cannot initiate other snapshots, the `reroute` API refuses to move the affected shard, and `_close`ing the index goes badly, because all shards report as unallocated (index=red), but the index will not fully close (nor re-open) because the shard is more or less 'locked' by the stuck snapshot.

The downtime from a full restart is unacceptable in production so this is currently preventing me from being able to use snapshots there, but I have a non-production cluster currently exhibiting the problem and I'll leave it in this 'stuck' state for a bit so I can answer any questions or produce API outputs, etc.

I uploaded some relevant output (/_shapshot/repo/_all, /_snapshot/repo/_status, /index/_stats?level=shards) here: https://gist.github.com/allthedrones/898f282be33a33dfec67

**Comment by imotov** [comment]
@allthedrones could you post the snapshot section of `_cluster/state`?

**Comment by allthedrones** [comment]
Here's the snapshot section of `_cluster/state`. I verified that the other two url endpoints under `_snapshot` still return the same results as posted earlier, and that attempting DELETE on the snapshot still hangs. Strange that all 3 places report slightly different statuses for the snapshot!

    "snapshots" : [{
        "repository" : "my_stuff",
        "snapshot" : "idx-15923",
        "include_global_state" : false,
        "state" : "ABORTED",
        "indices" : [ "idx-15923" ],
        "shards" : [ {
          "index" : "idx-15923",
          "shard" : 3,
          "state" : "SUCCESS",
          "node" : "cvnYD0WlQEi9-AmEiHTC4g"
        }, {
          "index" : "idx-15923",
          "shard" : 4,
          "state" : "SUCCESS",
          "node" : "rloJ197jS9eSWiDqy29QHA"
        }, {
          "index" : "idx-15923",
          "shard" : 1,
          "state" : "ABORTED",
          "node" : "q-F03vzxTBqdbUojCWeY9g"
        }, {
          "index" : "idx-15923",
          "shard" : 2,
          "state" : "SUCCESS",
          "node" : "KFVhLk0eRDmuGxLzZ1HEww"
        }, {
          "index" : "idx-15923",
          "shard" : 0,
          "state" : "SUCCESS",
          "node" : "FZ8pwJqsTgiz2rvDCBVrgw"
        } ]
    }]

**Comment by imotov** [comment]
@allthedrones is the node `q-F03vzxTBqdbUojCWeY9g` still part of the cluster? If it is, could you run the following command and post here the result 

`curl localhost:9200/_nodes/q-F03vzxTBqdbUojCWeY9g/hot_threads?threads=1000`

**Comment by allthedrones** [comment]
@imotov Here's the `hot_threads` result. The last block looks relevant ... like maybe a stream read operation from the Azure repo connector is waiting indefinitely?

    ::: [mc2-ed6][q-F03vzxTBqdbUojCWeY9g][mc2-ed6][inet[/10.31.0.11:9300]]{master=false}
       Hot threads at 2016-01-07T13:40:47.042Z, interval=500ms, busiestThreads=1000, ignoreIdleThreads=true:
        
        3.9% (19.5ms out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][bulk][T#2]'
         2/10 snapshots sharing following 2 elements
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
        
        2.5% (12.3ms out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][bulk][T#3]'
         3/10 snapshots sharing following 2 elements
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
        
        0.6% (2.8ms out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][bulk][T#4]'
         2/10 snapshots sharing following 10 elements
           org.elasticsearch.index.engine.InternalEngine.index(InternalEngine.java:371)
           org.elasticsearch.index.shard.IndexShard.index(IndexShard.java:511)
           org.elasticsearch.action.bulk.TransportShardBulkAction.shardIndexOperation(TransportShardBulkAction.java:413)
           org.elasticsearch.action.bulk.TransportShardBulkAction.shardOperationOnPrimary(TransportShardBulkAction.java:148)
           org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase.performOnPrimary(TransportShardReplicationOperationAction.java:574)
           org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase$1.doRun(TransportShardReplicationOperationAction.java:440)
           org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:36)
           java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
         unique snapshot
           sun.misc.Unsafe.park(Native Method)
           java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
           java.util.concurrent.LinkedTransferQueue.awaitMatch(LinkedTransferQueue.java:737)
           java.util.concurrent.LinkedTransferQueue.xfer(LinkedTransferQueue.java:647)
           java.util.concurrent.LinkedTransferQueue.take(LinkedTransferQueue.java:1269)
           org.elasticsearch.common.util.concurrent.SizeBlockingQueue.take(SizeBlockingQueue.java:162)
           java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
           java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
        
        0.5% (2.6ms out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][bulk][T#1]'
         3/10 snapshots sharing following 2 elements
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
        
        0.0% (205.7micros out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][transport_client_timer][T#1]{Hashed wheel timer #1}'
         10/10 snapshots sharing following 5 elements
           java.lang.Thread.sleep(Native Method)
           org.elasticsearch.common.netty.util.HashedWheelTimer$Worker.waitForNextTick(HashedWheelTimer.java:445)
           org.elasticsearch.common.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:364)
           org.elasticsearch.common.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
           java.lang.Thread.run(Thread.java:745)
        
        0.0% (0s out of 500ms) cpu usage by thread 'elasticsearch[mc2-ed6][snapshot][T#1]'
         10/10 snapshots sharing following 21 elements
           sun.misc.Unsafe.park(Native Method)
           java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
           java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
           java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
           java.util.concurrent.ExecutorCompletionService.take(ExecutorCompletionService.java:193)
           com.microsoft.azure.storage.blob.BlobOutputStream.waitForTaskToComplete(BlobOutputStream.java:456)
           com.microsoft.azure.storage.blob.BlobOutputStream.dispatchWrite(BlobOutputStream.java:348)
           com.microsoft.azure.storage.blob.BlobOutputStream.writeInternal(BlobOutputStream.java:578)
           com.microsoft.azure.storage.blob.BlobOutputStream.write(BlobOutputStream.java:506)
           com.microsoft.azure.storage.blob.BlobOutputStream.write(BlobOutputStream.java:482)
           com.microsoft.azure.storage.blob.BlobOutputStream.write(BlobOutputStream.java:541)
           org.elasticsearch.cloud.azure.blobstore.AzureOutputStream.write(AzureOutputStream.java:35)
           java.io.OutputStream.write(OutputStream.java:116)
           org.elasticsearch.index.snapshots.blobstore.BlobStoreIndexShardRepository$SnapshotContext.snapshotFile(BlobStoreIndexShardRepository.java:564)
           org.elasticsearch.index.snapshots.blobstore.BlobStoreIndexShardRepository$SnapshotContext.snapshot(BlobStoreIndexShardRepository.java:507)
           org.elasticsearch.index.snapshots.blobstore.BlobStoreIndexShardRepository.snapshot(BlobStoreIndexShardRepository.java:140)
           org.elasticsearch.index.snapshots.IndexShardSnapshotAndRestoreService.snapshot(IndexShardSnapshotAndRestoreService.java:85)
           org.elasticsearch.snapshots.SnapshotsService$5.run(SnapshotsService.java:871)
           java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)
        
        0.0% (0s out of 500ms) cpu usage by thread 'DestroyJavaVM'
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
         unique snapshot
        
        0.0% (0s out of 500ms) cpu usage by thread 'elasticsearch[keepAlive/1.6.1]'
         10/10 snapshots sharing following 8 elements
           sun.misc.Unsafe.park(Native Method)
           java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
           java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
           java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
           java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
           java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
           org.elasticsearch.bootstrap.Bootstrap$4.run(Bootstrap.java:267)
           java.lang.Thread.run(Thread.java:745)
        
        0.0% (0s out of 500ms) cpu usage by thread 'pool-3853-thread-1'
         10/10 snapshots sharing following 23 elements
           java.net.SocketInputStream.socketRead0(Native Method)
           java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
           java.net.SocketInputStream.read(SocketInputStream.java:170)
           java.net.SocketInputStream.read(SocketInputStream.java:141)
           java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
           java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
           java.io.BufferedInputStream.read(BufferedInputStream.java:345)
           sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:704)
           sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:647)
           sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1536)
           sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1441)
           java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
           com.microsoft.azure.storage.core.ExecutionEngine.executeWithRetry(ExecutionEngine.java:124)
           com.microsoft.azure.storage.blob.CloudBlockBlob.uploadBlockInternal(CloudBlockBlob.java:714)
           com.microsoft.azure.storage.blob.CloudBlockBlob.uploadBlock(CloudBlockBlob.java:686)
           com.microsoft.azure.storage.blob.BlobOutputStream$1.call(BlobOutputStream.java:362)
           com.microsoft.azure.storage.blob.BlobOutputStream$1.call(BlobOutputStream.java:358)
           java.util.concurrent.FutureTask.run(FutureTask.java:266)
           java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
           java.util.concurrent.FutureTask.run(FutureTask.java:266)
           java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
           java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
           java.lang.Thread.run(Thread.java:745)

**Comment by imotov** [comment]
@allthedrones indeed the 6th block and the last block show stuck azure operations. It looks like azure never came back with responses for these requests. I think this problem has been [fixed](#14277) for the next major release, but the only solution for 1.6.1 that I can think of is to restart the node where azure is stuck. 

@dadoonet, @tlrx what do you think?

**Comment by dadoonet** [comment]
I agree with your analysis @imotov. The azure client until #15080 wait indefinitely. There is no way to unblock it without restarting the node.

I wonder then if we should backport the change #15080 in 2.x branch.

**Comment by imotov** [comment]
@dadoonet I think it would be reasonable to backport it to 2.x branch considering the impact on the users.

**Comment by allthedrones** [comment]
Thanks guys, I can confirm that restarting the node with the stalled shard corrects the problem. After restarting that data node, the snapshot in question was fully deleted and all of the snapshot URLs mentioned previously report a correct & consistent status. Restarting a single data node is a *much* more acceptable impact than downtime for the whole cluster, so thank you again for your help arriving at a resolution & root cause.

**Comment by wangqinghuan** [comment]
I have the same problem when using NFS as repository, have this fixed  regarding  NFS repository? My Elasticsearch version is 6.1.3.
